profiling/variorum-connector/variorum-connector.cpp

```diff
@@ -226,27 +267,138 @@ void variorum_call_mpi() {
 // Description: The function determines what profiling options are selected and
 // prints the profoiling data out to std out Pre: None Post: An output message
 // if variourum returned an error or if it functioned correctly
-
 void variorum_call() {
-  std::string output;
+  // std::cout <<output  << std::endl;
+  long long temp = 0;
+
+  int num_sockets = 0;
+
   if (type_of_profiling == 0) {
-    output  = variorum_print_power_call();
-    char* s = variorum_json_call();
-    std::cout << s << "\n";
-    std::cout << output << std::endl;
+    output = variorum_print_power_call();
+    char *s = variorum_json_call();
+    // std::cout << s << "\n";
+    char *timestamp_pos = strstr(s, "\"timestamp\":");
+    if (timestamp_pos != nullptr) {
+      timestamp_pos += strlen("\"timestamp\":");
+      char *timestamp_end_pos = strchr(timestamp_pos, ',');
+      if (timestamp_end_pos == nullptr) {
+        timestamp_end_pos = strchr(timestamp_pos, '}');
+      }
+      if (timestamp_end_pos != nullptr) {
+        std::string timestamp_str(timestamp_pos,
+                                  timestamp_end_pos - timestamp_pos);
+        long long timestamp = std::stoll(timestamp_str);
+        // std::cout << "Timestamp: " << timestamp << std::endl;
+        if (power == -1 || time2 != 0) {
+          time1 = timestamp;
+          // std::cout << ""
+        } else {
+          time2 = timestamp;
+        }
+      }
```

Comment:
Can't you use jansson for parsing the json output?
Please add comments on what this code is doing.